TESTS/mbed_drivers/lp_timer/main.cpp

```diff
@@ -35,26 +35,21 @@ extern uint32_t SystemCoreClock;
  * timer we need to adjust delta.
  */
 
-/* Macro to define delta based on CPU clock frequency.
+ /*
+ * Define tolerance as follows:
+ * tolerance = 500 us + 5% of measured time
  *
- * Note that some extra time is counted by the timer.
- * Additional time is caused by the function calls and
- * additional operations performed by wait and
- * stop functions before in fact timer is stopped. This may
- * add additional time to the counted result.
+ * e.g.
+ * 1 ms delay: tolerance = 550 us
+ * 10 ms delay: tolerance = 1000 us
+ * 100 ms delay: tolerance = 5500 us
+ * 1000 ms delay: tolerance = 50500 us
  *
- * To take in to account this extra time we introduce DELTA
- * value based on CPU clock (speed):
- *   DELTA = TOLERANCE_FACTOR / SystemCoreClock * US_FACTOR
- *
- *   e.g.
- *   For K64F          DELTA = (80000 / 120000000) * 1000000 = 666[us]
- *   For NUCLEO_F070RB DELTA = (80000 /  48000000) * 1000000 = 1666[us]
- *   For NRF51_DK      DELTA = (80000 /  16000000) * 1000000 = 5000[us]
- *
- * As low power timer cannot be too much accurate, this DELTA should not be more precise than 500us,
- * which corresponds to a maximum CPU clock around 130MHz
- */
+ *  */
+#define DELTA_US(delay_ms) (500 + delay_ms * US_PER_MSEC / 20)
+#define DELTA_MS(delay_ms) (1 + (delay_ms * US_PER_MSEC / 20 / US_PER_MSEC))
+#define DELTA_S(delay_ms) (0.000500f + (((float)delay_ms) / MSEC_PER_SEC / 20))
```

Comment:
As macro in C (and c++) are basic text subtitution, it is usually good practice to wrap the macro's argument between brackets.
If you pass to `DELTA_US` an expression such as `a_var + something_else` you may not obtain what you expect.
```
#define DELTA_US(delay_ms) (500 + (delay_ms) * US_PER_MSEC / 20)
```

Also, as this is in test it does not really impact the final code but in debug mode, double division may not get optimised away and  `(1 + (delay_ms * US_PER_MSEC / 20 / US_PER_MSEC))` will try to run 2 divisions.

Reply:
From what I see on GH these three macros are defined twice (the other definitions are line 63).

Reply:
And worse than that they are actually defined differently!!

Reply:
Nice catch. I agree with the first part, but since arithmetic operations are not passed to the macro its not a critical error.
Unfortunately the second bug is unsafe. It looks like I forgot to remove old definitions 😢 .

Reply:
Can you add a commit fixing this?

Reply:
I don't know how I can add commit to `bulislaw:merge_feature_branches_for_5.9` branch.
Lines 49-51 should be inserted instead of code in lines to 59-65.

But the old version is not bed. It has been changed for consistency with other ticker tests.

Reply:
@bulislaw needs to add you as collaborator to his repo.

Reply:
I can help, got that branch locally, point me to a commit and I'll cherry pick and push

Reply:
@mprse actually there are arithmetic in [mbed_drivers/timer/main.cpp#L724](https://github.com/bulislaw/mbed-os/blob/085a3a08a2793b2302ac77292e63734c8aedbe2d/TESTS/mbed_drivers/timer/main.cpp#L724)
It is ok as it is just division but for maintainability it might create non-obvious bugs.

Reply:
Provided fix adds brackets.